torch_geometric/nn/conv/han_conv.py

```diff
@@ -0,0 +1,164 @@
+from typing import Union, Dict, Optional, List
+
+import torch
+from torch import Tensor, nn
+import torch.nn.functional as F
+
+from torch_geometric.typing import NodeType, EdgeType, Metadata
+from torch_sparse import SparseTensor
+from torch_geometric.nn.dense import Linear
+from torch_geometric.utils import softmax
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.inits import glorot, reset
+
+
+def group(xs: List[Tensor], q: nn.Parameter,
+          k_lin: nn.Module) -> Optional[Tensor]:
+    if len(xs) == 0:
+        return None
+    else:
+        num_edge_types = len(xs)
+        out = torch.stack(xs)
+        attn_score = (q * torch.tanh(k_lin(out)).mean(1)).sum(-1)
+        attn = F.softmax(attn_score, dim=0)
+        out = torch.sum(attn.view(num_edge_types, 1, -1) * out, dim=0)
+        return out
+
+
+class HANConv(MessagePassing):
+    r"""
+    The Heterogenous Graph Attention Operator from the
+    `"Heterogenous Graph Attention Network"
+    <https://arxiv.org/pdf/1903.07293.pdf>`_ paper.

```

Comment:
Shall we add a small example to show-case the usage of `HANConv` in combination with `T.AddMetaPaths()`?

Reply:
Yeah, I think that's needed. Will create an issue for that.